ssl.pl

```diff
@@ -78,6 +78,7 @@
 		       certificate_file(atom),
 		       key_file(atom),
 		       password(any),
+		       cipher_list(any),
```

Comment:
"Cipher list" is [SSL terminology](https://www.openssl.org/docs/manmaster/apps/ciphers.html) to denote ciphers in a special syntax. In all the uses I have seen, it is an opaque object that has a special meaning within OpenSSL. It is typically used in configuration files and rather static. 

Please see the aptly named [**https://cipherli.st/**](https://cipherli.st/) web site for an overview of the different syntax options that other systems provide. A name somehow involving "list" is still one of the most frequently occurring common names for this option (the other 6 systems use a name that just ends with `ciphers`, but these options do not have the same syntax across the systems that use it):

<pre>
*cipher[-_]?list :  <b>3</b> (Lighthttpd, Postfix, Dovecot)
*CipherSuite:       <b>2</b> (Apache, ProFTPd)
</pre>
I suggest to also call this `cipher_list` and use an opaque Prolog atom for configuration, since this allows us to freely exchange the configurations between the different systems. Much of the available advice regarding ciphers is exchanged in the form of such strings, and deviating from this syntax without good reason would only introduce an additional source of potential user errors and security problems. For example, I got a quite nice security assessment using the HTTP server with `--cipherlist='DEFAULT:!RC4:!DES-CBC-SHA'`, and there are many other options suggested by the community that make sense and which you can try easily due to this syntax compatibility with other systems.

If there ever is a need to assemble this option dynamically, we can still cover this use case with an option named `ciphers/1`; also, using `atomic_list_concat/3`, it is very easy to assemble such an atom dynamically.

@chigley: Please comment if you have any input. Thank you!